src/main/scala/abalone/Pos.scala

```diff
@@ -324,17 +325,74 @@ object Pos {
     }
   }
 
-  def possibleLineDirsFromSideMoveDir(dir: Direction): Directions = {
+  def potentialLineDirsFromSideMoveDir(sideMoveDir: Direction): Directions = {
     val x = Pos.E5
-    val y = dir(x).getOrElse(Pos.A1)
-    x.dir(y) match {
+    val y = sideMoveDir(x).getOrElse(Pos.A1)
+    x.directionString(y) match {
+      case "upLeft"     => List(_.left, _.upLeft)
+      case "upRight"    => List(_.upLeft, _.upRight, _.right)
+      case "downRight"  => List(_.right, _.downRight)
+      case "downLeft"   => List(_.downRight, _.downLeft, _.left)
+      case _            => List()
+    }
+  }
+  def potentialSideMoveDirsFromGlobalDir(globalDir: Direction): Directions = {
+    val x = Pos.E5
+    val y = globalDir(x).getOrElse(Pos.A1)
+    x.directionString(y) match {
       case "upLeft"     => List(_.left, _.upLeft)
-      case "upRight"    => List(_.upRight, _.right)
+      case "upRight"    => List(_.upLeft, _.upRight, _.right)
       case "downRight"  => List(_.right, _.downRight)
-      case "downLeft"   => List(_.downLeft, _.left)
+      case "downLeft"   => List(_.downLeft, _.left, _.downRight)
       case _            => List()
     }
   }
+  def deducePotentialSideDirs(globalDir: Direction, lineDir: Direction): Directions = {
+    val x = Pos.E5
+    val y = globalDir(x).getOrElse(Pos.A1)
+    val z = lineDir(x).getOrElse(Pos.A1)
+    x.directionString(y) match {
+      case "downLeft" => {
+        x.directionString(z) match {
+          case "downLeft" => List(_.left, _.downRight)
+          case "downRight" => List(_.downLeft)
+          case "left" => List(_.downLeft)
+        }
+      }
+      case "upRight" => {
+        x.directionString(z) match {
+          case "upLeft" => List(_.upRight)
+          case "upRight" => List(_.right, _.upLeft)
+          case "right" => List(_.upRight)
+        }
+      }
+      case "upLeft" => {
+        x.directionString(z) match {
+          case "upLeft" => List(_.left)
+          case "left" => List(_.upLeft)
+        }
+      }
+      case "downRight" => {
+        x.directionString(z) match {
+          case "downRight" => List(_.right)
+          case "right" => List(_.downRight)
+        }
+      }
+    }
+  }
+
+  def reverseDir(dir: Direction): Direction = {
+    val x = Pos.E5
+    val y = dir(x).getOrElse(Pos.A1)
+    x.directionString(y) match {
+      case "left"       => _.right
+      case "upLeft"     => _.downRight
+      case "upRight"    => _.downLeft
+      case "right"      => _.left
+      case "downRight"  => _.upLeft
+      case "downLeft"   => _.upRight
+    }
+  }
```

Comment:
Is this equivalent to:
```
def reverseDir(dir: Direction): Direction =
dir match {
  case _.left => _.right
  case _.right => _.left
  case _.upLeft => _.downRight
  case _.upRight => _.downLeft
  case _.downLeft => _.upRight
  case _.downRight => _.upLeft
}
```
I think it is and copilot thinks so too
It might even be possible to ditch the whole string based idea for directions and use the true `Direction` type?